nuxeo-routing-core/src/main/java/org/nuxeo/ecm/platform/routing/core/impl/DocumentRoutingServiceImpl.java

```diff
@@ -1460,4 +1483,67 @@ public void run() {
     public boolean isWorkflowModel(final DocumentRoute documentRoute) {
         return documentRoute.isValidated();
     }
+
+    /**
+     * We need this listener due to the {@code unstash} reload strategy which doesn't call repository initializator.
+     *
+     * @since 9.2
+     */
+    public static class DocumentRoutingRuntimeListener implements ComponentListener {
+
+        @Override
+        public void handleEvent(ComponentEvent event) {
+            // only handle event for component registering routes
+            if (event != null && ComponentEvent.COMPONENT_STARTED == event.id) {
+                RegistrationInfo ri = event.registrationInfo;
+                // look for this service registration
+                boolean hasRouteContribution = Stream.of(ri.getExtensions())
+                                                     .map(Extension::getTargetComponent)
+                                                     .map(ComponentName::getName)
+                                                     .anyMatch("org.nuxeo.ecm.platform.routing.service"::equals);
+                if (hasRouteContribution) {
+                    importAllRouteModels();
+                }
+            }
+        }
+
+        protected void importAllRouteModels() {
+            // This code is welly done by RouteModelsInitializator, but in development mode we also want to execute it
+            // during hotreload. Currently "unstash" reload strategy doesn't allow to correctly do that, once we'll have
+            // changed the strategy, initializators will be called after a hotreload
+            if (!Framework.isDevModeSet()) {
+                log.info("Do not import all route models: dev mode is not set");
+                return;
+            }
+            try {
+                RepositoryManager rm = Framework.getService(RepositoryManager.class);
+                // Transaction management
+                boolean txStarted = !TransactionHelper.isTransactionActive() && TransactionHelper.startTransaction();
+                boolean txSucceed = false;
+                try {
+                    new UnrestrictedSessionRunner(rm.getDefaultRepositoryName()) {
```

Comment:
Why don't you use `TransactionHelper.runInTransaction` and `CoreInstance.openCoreSession`?